sys/include/net/nanocoap_sock.h

```diff
@@ -7,7 +7,109 @@
  */
 
 /**
- * @ingroup     net_nanocoap
+ * @defgroup    net_nanosock Nanocoap Sock
+ * @ingroup     net
+ * @brief       Synchronous sock based messaging with nanocoap
+ *
+ * nanocoap sock uses the nanocoap CoAP library to provide a synchronous
+ * interface to RIOT's sock networking API to read and write CoAP messages.
+ * For a server, nanocoap sock accepts a list of resource paths with callbacks
+ * for writing the response. For a client, nanocoap sock provides a function
+ * to send a request and waits for the server response.
+ *
+ * ## Server Operation ##
+ *
+ * See the nanocoap_server example, which is built on the nanocoap_server()
+ * function. A server must define an array of coap_resource_t resources for
+ * which it responds. See the declarations of `coap_resources` and
+ * `coap_resources_numof`. The array contents must be ordered by the resource
+ * path, specifically the ASCII encoding of the path characters (digit and
+ * capital precede lower case). nanocoap provides the
+ * COAP_WELL_KNOWN_CORE_DEFAULT_HANDLER entry for `/.well-known/core`.
+ *
```

Comment:
The `path matching` section is missing here.

Reply:
Good catch, Leandro!

I left the path matching section in the base nanocoap doc and added references in the nanocoap sock and gcoap docs. There was too much text in that section to duplicate.